tensorflow/compiler/mlir/lite/transforms/optimize.cc

```diff
@@ -198,7 +198,7 @@ DenseElementsAttr GetShape(Value output_val) {
   auto output_type = output_val.getType().cast<RankedTensorType>();
   auto shape_vector = output_type.getShape();
   std::vector<int32_t> shape(shape_vector.size());
-  for (int i = 0; i < shape_vector.size(); ++i) {
+  for (int i = 0, end = shape_vector.size(); i < end; ++i) {
     shape[i] = shape_vector[i];
   }
```

Comment:
This loop should be written using range loops and `emplace_back`/`push_back`.

Reply:
updated.